auth_server.py

```diff
@@ -68,11 +68,15 @@ def auth():
         else:
             auth_result_code = 200
     except BadSignature as e:
+        vm_ip = None
         auth_result_code = 401
-    return vm_ip, int(auth_result_code)
+
+    headers = {'X-Target-VM-IP': vm_ip}
+
+    return (vm_ip, int(auth_result_code), headers)
```

Comment:
Thinking about it, we probably only need to return the vm_ip and and headers if auth_result_code == 200.